src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -398,13 +415,83 @@ def acr_task_update(cmd,  # pylint: disable=too-many-locals
     return client.update(resource_group_name, registry_name, task_name, taskUpdateParameters)
 
 
+def acr_task_identity_assign(cmd,
+                             client,
+                             task_name,
+                             registry_name,
+                             identities=None,
+                             resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    identity = _build_identities_info(cmd, identities)
+
+    TaskUpdateParameters = cmd.get_models('TaskUpdateParameters')
+
+    taskUpdateParameters = TaskUpdateParameters(
+        identity=identity
+    )
+
+    return client.update(resource_group_name, registry_name, task_name, taskUpdateParameters)
+
+
+def acr_task_identity_remove(cmd,
+                             client,
+                             task_name,
+                             registry_name,
+                             identities=None,
+                             resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    if identities is not None and IDENTITY_GLOBAL_REMOVE in identities:
+        if len(identities) > 1:
+            raise CLIError(
+                "Cannot specify additional identities when [all] is used in [--identities]")
+
+    identity = None
+    if not identities or IDENTITY_LOCAL_ID in identities:
+        # To remove only the system assigned identity if user-assigned identities also exist
+        # PATCH with the existing user-assigned identities
+        # If no user-assigned identities exist, set the type to None
+        existingUserIdentites = client.get_details(
+            resource_group_name, registry_name, task_name).identity.user_assigned_identities
```

Comment:
Will `get_details().identity` be None?

Reply:
Good catch. Adding a null check here in case the user tries to remove the system identity when Identity is null for the task